pytorch_lightning/callbacks/pruning.py

```diff
@@ -415,13 +433,8 @@ def on_save_checkpoint(
     ) -> Dict[str, Any]:
         if self._make_pruning_permanent:
             rank_zero_debug("`ModelPruning.on_save_checkpoint`. Pruning is made permanent for this checkpoint")
-            prev_device = pl_module.device
-            # prune a copy so training can continue with the same buffers
-            copy = deepcopy(pl_module.to("cpu"))
-            self.make_pruning_permanent(copy)
```

Comment:
`make_pruning_permanent` handles the pruning forward hooks, not just the mask multiplication.

Are you sure not using it is a good idea? Don't you think we could be missing special logic by handling the pruning ourselves?

Reply:
Here is what the remove function does. This is not required as we keep training and we can just do it manually.

```py
 def remove(self, module):
        r"""Removes the pruning reparameterization from a module. The pruned
        parameter named ``name`` remains permanently pruned, and the parameter
        named ``name+'_orig'`` is removed from the parameter list. Similarly,
        the buffer named ``name+'_mask'`` is removed from the buffers.

        Note:
            Pruning itself is NOT undone or reversed!
        """
        # before removing pruning from a tensor, it has to have been applied
        assert (
            self._tensor_name is not None
        ), "Module {} has to be pruned\
            before pruning can be removed".format(
            module
        )  # this gets set in apply()

        # to update module[name] to latest trained weights
        weight = self.apply_mask(module)  # masked weights

        # delete and reset
        if hasattr(module, self._tensor_name):
            delattr(module, self._tensor_name)
        orig = module._parameters[self._tensor_name + "_orig"]
        orig.data = weight.data
        del module._parameters[self._tensor_name + "_orig"]
        del module._buffers[self._tensor_name + "_mask"]
        setattr(module, self._tensor_name, orig)
```